### PR TITLE
Semi-automated dataset creation for Amazon Robotics Challenge 2017

### PR DESCRIPTION
**Pull request by wkentaro**
```
baxter@sheeta $ source ~/ros/wkentaro_ws/devel/setup.zsh
```

## Semi-automated raw data collection

```
roslaunch jsk_2016_01_baxter_apc baxter.launch
roslaunch jsk_2016_01_baxter_apc astra_hand.launch left_first:=true  # maybe false

# pack the shelf bins
roslaunch jsk_arc2017_baxter collect_data_in_bin.launch  # data are saved to ~/.ros/jsk_arc2017_baxter/dataset_jsk_v1
```

## Annotation

```
# ln -s ~/.ros/jsk_arc2017_baxter/collect_data_in_bin $(rospack find jsk_arc2017_common)/data/datasets/JSK_V1
rosrun jsk_arc2017_common annotate_dataset_v1.py
```

## Tutorial Movie
https://drive.google.com/open?id=0B9P1L--7Wd2vNVZPdnpzdllRamc
<a href="https://drive.google.com/open?id=0B9P1L--7Wd2vNVZPdnpzdllRamc">screen shot 2017-03-17 at 11 03 34</a>

## Appendix

how to generate cleaned data from training data published by Amazon Robotics?

```
ln -s XXX/Trainng items $(rospack find jsk_arc2017_common)/data/datasets/AR_20170224
rosrun jsk_arc2017_common parse_ar_dataset.py
rosrun jsk_arc2017_common vis_object_list.py
```

### CODE REVIEW

File: jsk_arc2017/common/package.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0"?>
+<package>
+  <name>jsk_arc2017_common</name>
+  <version>2.0.0</version>
```
Comment (knorth55) [inline jsk_arc2017/common/package.xml]: 0.0.1?

File: jsk_arc2017/baxter/package.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0"?>
+<package>
+  <name>jsk_arc2017_baxter</name>
+  <version>2.0.0</version>
```
Comment (knorth55) [inline jsk_arc2017/baxter/package.xml]: 0.0.1?

**Comment by knorth55** [comment]
We ahould think twice and discuss about pacakge name, description and dir trees.

**Comment by knorth55** [comment]
I'm for the package names not dir trees.
jsk_arc2017_common should be one dir as jsk_apc2016_common.

File: jsk_arc2017/common/package.xml (hunk 1 above)
Comment (wkentaro) [inline jsk_arc2017/common/package.xml]: no it should corresponds to other packages.

Comment (knorth55) [inline jsk_arc2017/common/package.xml]: ok

**Comment by wkentaro** [comment]
Waiting for https://github.com/start-jsk/jsk_apc/issues/2027

File: jsk_arc2017/baxter/launch/collect_data_in_bin.launch [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    <rosparam subst_value="true">
+      save_dir: ~/.ros/jsk_arc2017_baxter/dataset_jsk_v1
+      topics:
+        - name: /left_hand_camera/rgb/image_rect_color
+          msg_class: sensor_msgs/Image
+          fname: image.png
+          savetype: ColorImage
+        - name: /left_hand_camera/depth_registered/hw_registered/image_rect_raw
+          msg_class: sensor_msgs/Image
+          fname: depth.pkl
```
Comment (wkentaro) [inline jsk_arc2017/baxter/launch/collect_data_in_bin.launch]: @otsubo Maybe you can try this program for your task.

**Comment by wkentaro** [comment]
Renamed the packages.

**Comment by wkentaro** [comment]
Depends on  https://github.com/jsk-ros-pkg/jsk_common/pull/1509
image

**Pull request merged**